application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java

```diff
@@ -132,12 +132,18 @@ public DesignatedBodyDto getDesignatedBodyCode(final String gmcId) {
     return DesignatedBodyDto.builder().designatedBodyCode(designatedBodyCode).build();
   }
 
-  public void removeDesignatedBodyCode(final ConnectionMessageDto message) {
+  public void updateDesignatedBodyCode(final ConnectionMessageDto message) {
     final var doctorsForDBOptional = doctorsRepository.findById(message.getGmcId());
     if (doctorsForDBOptional.isPresent()) {
       log.info("Updating designated body code from doctors for DB");
+      final var dbc = message.getDesignatedBodyCode();
       final var doctorsForDB = doctorsForDBOptional.get();
-      doctorsForDB.setDesignatedBodyCode(message.getDesignatedBodyCode());
+      doctorsForDB.setDesignatedBodyCode(dbc);
+      if (dbc == null) {
```

Comment:
Checked ConnectionService, it seems the DBC in the message is the previous DBC instead of the new current DBC (https://github.com/Health-Education-England/tis-revalidation-connection/blob/f46e0ff1181e96930336da7df189bc3cf765c5ad/src/main/java/uk/nhs/hee/tis/revalidation/connection/service/ConnectionService.java#LL230C40-L230C58).
I might be wrong, so wondering if you have checked this in the messages.